main.cpp

```diff
@@ -61,124 +69,411 @@ New/This/Pointers/References conclusion
+    FloatType& subtract ( const IntType& );
+    FloatType& multiply ( const IntType& );
+    FloatType& divide ( const IntType& );
 };
 
-float FloatType::add (float x, float y)
+FloatType& FloatType::add ( const float x )
 {
-    return (x+y);
+    *pointerToFloatValue += x;
+    return *this;
 }
 
-float FloatType::subtract (float x, float y) 
+FloatType& FloatType::subtract ( const float y ) 
 { 
-    return (x-y); 
+    *pointerToFloatValue -= y;
+    return *this; 
 }
 
-float FloatType::multiply (float x, float y) 
+FloatType& FloatType::multiply ( const float y) 
 { 
-    return (x*y); 
+    *pointerToFloatValue *= y;
+    return *this; 
 }
 
-float FloatType::divide (float x, float y) 
+FloatType& FloatType::divide ( const float y ) 
 { 
-    if (abs(y) > 0) return (x/y);    
+    if (abs(y) > 0) *pointerToFloatValue /= y;    
     
     std::cout << "Divide-by-zero warning  " << std::endl;
-    return (x/y);      
+    *pointerToFloatValue /= 0;
+    return *this;      
 }
 
 struct DoubleType
 {
-    double add (double, double);
-    double subtract ( double, double);
-    double multiply ( double, double);
-    double divide (double, double);
+    double* pointerToDoubleValue = nullptr;
+
+    // default constructor assigns heap primitive if no value sent in
+    DoubleType() {
+        pointerToDoubleValue = new double();
+    } 
+
+    DoubleType(double doubleIn )
+    {
+        pointerToDoubleValue = new double( doubleIn );
+    }
+    
+    ~DoubleType()
+    {
+        delete pointerToDoubleValue;
+        pointerToDoubleValue = nullptr;
+    }
+
+    DoubleType& add ( double );
+    DoubleType& subtract ( double );
+    DoubleType& multiply ( double );
+    DoubleType& divide ( double );
+    DoubleType& add ( const FloatType& );
+    DoubleType& subtract ( const FloatType& );
+    DoubleType& multiply ( const FloatType& );
+    DoubleType& divide ( const FloatType& );
+    DoubleType& add ( const IntType& );
+    DoubleType& subtract ( const IntType& );
+    DoubleType& multiply ( const IntType& );
+    DoubleType& divide ( const IntType& );
 };
 
-double DoubleType::add (double x, double y)
+DoubleType& DoubleType::add ( const double y)
 {
-    return (x+y);
+    *pointerToDoubleValue += y;
+    return *this;
 }
 
-double DoubleType::subtract (double x, double y) 
+DoubleType& DoubleType::subtract ( const double y) 
 { 
-    return (x-y); 
+    *pointerToDoubleValue -= y;
+    return *this; 
 }
 
-double DoubleType::multiply (double x, double y) 
+DoubleType& DoubleType::multiply ( const double y) 
 { 
-    return (x*y); 
+    *pointerToDoubleValue *= y;
+    return *this; 
 }
 
-double DoubleType::divide (double x, double y) 
+DoubleType& DoubleType::divide ( const double y) 
 { 
-    if (abs(y) > 0) return (x/y);    
+    if (abs(y) > 0) *pointerToDoubleValue /= y;    
     
     std::cout << "Divide-by-zero warning  " << std::endl;
-    return (x/y);      
+    *pointerToDoubleValue /= 0;
```

Comment:
you need an `else` here.  you're always dividing by 0. 

- [ ] add `else` block, remove division by zero.  floating point division by zero is allowed.